src/split.c

```diff
@@ -118,7 +118,7 @@ rcutils_split(
   }
 
   RCUTILS_SET_ERROR_MSG("unable to allocate memory for string array data");
-  return RCUTILS_RET_ERROR;
+  return RCUTILS_RET_BAD_ALLOC;
```

Comment:
@christophebedard line 105-106 (last trailing token) should be checked if that allocation fails, that also should go to `RCUTILS_RET_BAD_ALLOC`?

Reply:
Yep! I noticed it as well in our internal code. I'll create a PR soon.

Reply:
> Yep! I noticed it as well in our internal code. I'll create a PR soon.

It might be worthwhile to just refactor it to solve the TODO on line 67, which I think will also mostly handle this case.